includes/openid-connect-generic-login-form.php

```diff
@@ -171,21 +181,28 @@ function make_error_output( $error_code, $error_message ) {
 	 * @return string
 	 */
 	function make_login_button( $atts = array() ) {
-		$button_text = __( 'Login with OpenID Connect', 'daggerhart-openid-connect-generic' );
-		if ( ! empty( $atts['button_text'] ) ) {
-			$button_text = $atts['button_text'];
-		}
 
-		$text = apply_filters( 'openid-connect-generic-login-button-text', $button_text );
+		$atts = shortcode_atts(
+			array(
+				'button_text' => __( 'Login with OpenID Connect', 'daggerhart-openid-connect-generic' ),
+				'redirect_to' => $this->get_redirect_to(),
+			),
+			$atts,
+			'openid_connect_generic_login_button'
+		);
+
+		$text = apply_filters( 'openid-connect-generic-login-button-text', $atts['button_text'] );
+
 		$href = $this->client_wrapper->get_authentication_url( $atts );
 
-		ob_start();
-		?>
-		<div class="openid-connect-login-button" style="margin: 1em 0; text-align: center;">
-			<a class="button button-large" href="<?php print esc_url( $href ); ?>"><?php print $text; ?></a>
-		</div>
-		<?php
-		return wp_kses_post( ob_get_clean() );
```

Comment:
I don't see anything else that is escaping the login button text, so I think we want to keep this wp_kses_post().

Reply:
@daggerhart so actually that seemed to be causing some issues with the output, and actually all of the WordPress Coding Standards & Static Analysis checks passed. Looking at this again however I'm not escaping the `$text` & `$href` variables any more which could actually be opened to abuse by the use of the hooks. I'll check this again as I was seeing issues with the output that was related to once of the issues opened.

Reply:
@daggerhart I've added escaping to the `$text` & `$href` variables. I can't use `wp_kses_post()` otherwise it incorrectly re-formats the supplied button `href` URL.